packages/networking/test/unit/agent_spec.ts

```diff
@@ -1,181 +1,264 @@
+            }).on('message', resolve)
+          })
+          .then(msg => {
             expect(msg).to.eq('It worked!')
             if (this.debugProxy) {
               expect(this.debugProxy.requests[0].ws).to.be.true
               expect(this.debugProxy.requests[0].url).to.include('http://localhost:31080')
             }
-            resolve()
           })
         })
-      })
 
-      it('HTTPS websocket connections can be established and used', function() {
-        return new Promise((resolve) => {
-          Io.client(`https://localhost:${HTTPS_PORT}`, {
-            agent: this.agent,
-            transports: ['websocket']
-          }).on('message', (msg) => {
+        it('HTTPS websocket connections can be established and used', function() {
+          return new Promise((resolve) => {
+            Io.client(`https://localhost:${HTTPS_PORT}`, {
+              agent: this.agent,
+              transports: ['websocket']
+            }).on('message', resolve)
+          })
+          .then(msg => {
             expect(msg).to.eq('It worked!')
             if (this.debugProxy) {
               expect(this.debugProxy.requests[0]).to.include({
                 url: 'localhost:31443'
               })
             }
-            resolve()
           })
         })
       })
     })
   })
+
+  context("#_buildConnectReqHead", function() {
+    it('builds the correct request', function() {
+      const head = agent._buildConnectReqHead('foo.bar', '1234', {})
+      expect(head).to.eq([
+        'CONNECT foo.bar:1234 HTTP/1.1',
+        'Host: foo.bar:1234',
+        '', ''
+      ].join('\r\n'))
+    })
+
+    it('can do Proxy-Authorization', function() {
+      const head = agent._buildConnectReqHead('foo.bar', '1234', {
+        auth: 'baz:quux'
+      })
+      expect(head).to.eq([
+        'CONNECT foo.bar:1234 HTTP/1.1',
+        'Host: foo.bar:1234',
+        'Proxy-Authorization: basic YmF6OnF1dXg=',
+        '', ''
+      ].join('\r\n'))
+    })
+  })
+
+  context("#_createProxySock", function() {
+    it("creates a `net` socket for an http url", function() {
+      sinon.stub(net, 'connect')
+      const proxy = url.parse('http://foo.bar:1234')
+      agent._createProxySock(proxy)
+      expect(net.connect).to.be.calledWith(1234, 'foo.bar')
+    })
+
+    it("creates a `tls` socket for an https url", function() {
+      sinon.stub(tls, 'connect')
+      const proxy = url.parse('https://foo.bar:1234')
+      agent._createProxySock(proxy)
+      expect(tls.connect).to.be.calledWith(1234, 'foo.bar')
+    })
+
+    it("throws on unsupported proxy protocol", function() {
+      const proxy = url.parse('socksv5://foo.bar:1234')
+      try {
+        agent._createProxySock(proxy)
+        throw new Error("Shouldn't be reached")
+      } catch (e) {
+        expect(e.message).to.eq("Unsupported proxy protocol: socksv5:")
+      }
+    })
+  })
+
+  context("#_isResponseStatusCode200", function() {
```

Comment:
this is an exported function, not an instance method on a class - so it should use the `.` not the `#`.

`#` typically implies it being on the prototype of a class